### PR TITLE
Add implementation to TextProcessorJDK.topVowelAndConsonant() for haiku-kata-solutions module

### LINKED ISSUE #296: Add implementation to TextProcessorJDK.topVowelAndConsonant() for haiku-kata-solutions module
The method `topVowelAndConsonant()` in the `TextProcessorJDK` class for the [haiku-kata-solutions](https://github.com/eclipse/eclipse-collections-kata/tree/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions) module is not yet implemented. Follow the TODOs and hints given in the [method](https://github.com/eclipse/eclipse-collections-kata/blob/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java#L95). Please do not add solutions to the non-solution modules. After writing the code, uncomment the test and tag annotations and check that the [test case](https://github.com/eclipse/eclipse-collections-kata/blob/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions/src/test/java/org/eclipse/collections/haikukata/TextProcessorJDKTest.java#L56) `topVowelAndConsonant()` in the `TextProcessorJDKTest` passes for the [haiku-kata-solutions](https://github.com/eclipse/eclipse-collections-kata/tree/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions) module. Make sure to adhere to our [contribution guide](https://github.com/eclipse/eclipse-collections-kata/blob/52070f7651667c27e974e45eea2cce4309345530/CONTRIBUTING.md). 

Please note that this issue is created for GHC 2023 Open Source Day, thus priority will be given to the attendees of this event. After the event, this issue will be open to anyone.

### PR DESCRIPTION
**Pull request by rsp2210**
Fix #296

### CODE REVIEW

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 1]
```diff
@@ -96,24 +96,34 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
...
-
-        // TODO: Find the top consonant
-        // Hint: Use the filter method with findFirst, orElseThrow on Stream with the isVowel method below to find
-        // Hint: the top consonant Character value using getKey on Map.Entry.
-        char topConsonant = 'b';
+    public TopVowelAndConsonant topVowelAndConsonant() {
+        // Find all of the alphabetic letters from the haiku, convert them to lowercase,
+        // and count the Character values in a Map.
+        Map<Character, Long> charCounts = this.getHaikuAsChars()
+                .stream()
```
Comment (rachanavishwanath) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:103]: can you check we can use .select instead?

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 2]
```diff
@@ -96,24 +96,34 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
...
+                .filter(Character::isAlphabetic)
```
Comment (rachanavishwanath) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:104]: can we use .select here?

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 3]
```diff
@@ -96,24 +96,34 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
...
+
+        // Create a List of the top 26 occurrences in Map.Entry<Character, Long> instances sorted in reverse order by count.
+        List<Map.Entry<Character, Long>> entries = charCounts.entrySet()
+                .stream()
+                .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                .toList();
+
+        // Find the top vowel.
+        char topVowel = entries.stream()
+                .filter(this::isVowel)
```
Comment (rzrobin213) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:116]: You need to implement isVowel too

**Review by rzrobin213 — CHANGES_REQUESTED** [review]

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java (hunk 3 above)
Comment (rsp2210) [reply haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:116]: I implemented the function

Comment (rzrobin213) [reply haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:116]: do you have to call .getKey() for isVowel?

**Review by rzrobin213 — APPROVED** [review]

**Review by rachanavishwanath — COMMENTED** [review]
Please squash your commits

**Pull request closed without merging**